tests/core/test_composition.py

```diff
@@ -141,6 +141,10 @@ def test_init(self):
         comp = Composition({"S": Composition.amount_tolerance / 2})
         assert len(comp.elements) == 0
 
+        # test from int/float
+        assert Composition(1) == Composition("H")
+        assert Composition(2.0) == Composition("He")
```

Comment:
curious if anyone has thoughts on if we really want this behavior and if so, can you think of more edge cases to test. @JaGeo @Andrew-S-Rosen @shyuep @DanielYang59

Reply:
Mh. I feel like in the composition case, this might be a bit too much as a composition would also include floats and ints to specify the exact composition.

Reply:
For species, I agree that this would be useful but I am not so sure about composition.

Reply:
@JaGeo thanks for chiming in!

i should have mentioned the initial motivation for this change came from me noticing (and being surprised) that `Structure` already handles `int` (e.g. `si = 14` below):

https://github.com/materialsproject/pymatgen/blob/48af0386a80b5a6e59d97b42f62529358c43cbb3/tests/io/vasp/test_inputs.py#L383-L392

i'm not so much worried about the code getting messy. a single `int`/`float` was easy to handle in the `Composition` constructor. i'm more worried user expectations for what `Composition(1)` should return could be inconsistent? but then i can't see what other behavior one might expect